src/session/error.rs

```diff
@@ -91,6 +91,8 @@ pub enum ParseError {
     Malformed(Box<dyn std::error::Error + Send + Sync>),
     /// An error occurred when decoding bytes of a proto message.
     Decode(prost::DecodeError),
+    /// An error occurred when parsing Vec<u8> to Regex.
+    RegexParse(RegexParseError)
```

Comment:
I think this error is too specific to make it a separate value for the enum. `Malformed` should cover this case.